doc/proposals/hardware-plugin/pins-api/unify-pins-interconnections.md

```diff
@@ -0,0 +1,221 @@
+
+```yaml
+e810:
+	interconnections:
+	- id: ens5f0
+		part: E810-XXVDA4T
+		inputPhaseDelay:
+			connector: SMA1
+		delayPs: 920
+		phaseOutputConnectors:
+		- SMA2
+	- id: ens4f0
+		part: E810-XXVDA4T
+		gnssInput: false
+		phaseOutputConnectors:
+		- SMA1
+		- SMA2
+	- id: ens8f0
+		part: E810-XXVDA4T
+		inputPhaseDelay:
+			connector: SMA1
+		delayPs: 920
+		phaseOutputConnectors:
+		- SMA2
+
+```   
+
+There are multiple problems with the current solution for pin configuration:
+1. Both APIs are a part of the `e810` hardware plugin. To support other manufacturers this code will need to be duplicated for every manufacturer
+2. The interconnections API contains everything currently included in the pin API, so using both APIs could be seen as duplication
+3. The internal delays are hardcoded per part number. This might create a maintenance burden to extend them for multiple cards and keep these up-to-date. 
+
+### User Stories
+- As a user I want to avoid duplication of hardware configurations, so my configuration will be more compact and less error-prone
+- As a user I want to be able to use NICs of several manufacturers interconnected in the same system, so I can reach my business goals
+- As a user I want to use a single API that will allow me configuring all the aspects of frequency and phase interfaces, so I can configure the supporting I/O to be frequency, phase or eSync input or output and build a flexible multi-NIC topology.
+- As a user I want to be able to override the hardcoded internal delay numbers, so I can configure the internal delays for a new card, or adapt existing settings to my custom environment
+- As a user I want the combination of APIs verified for validity and the system fails in case of error, so I can fix the error fast
+
+### Goals
+
+- Ensure the interconnections API is suitable for T-BC
+- Make pins API optional if the interconnections API is configured.
+  - If both APIs are configured, use the pins API to configure the E-810 through the driver, and interconnections API to configure the delays through the DPLL API.
+- Provide a new API for overriding the hardcoded cards and their internal delays with user-defined cards and delays.  
+
+### Non-Goals
+To be considered for future development:
+- Modification of current T-GM functionality
+- Support of non-Intel hardware plugins or plugins other than e810 (file a separate EP)
+- Support of 10MHz / eSync on the pins (Check if needed and file a separate EP)
+
+## Proposal
+1. Add an option for to specify generic cards with non-default internal delays
+2. Make the pins API optional
+3. Verify for consistency
+
+
+### Workflow Description
+
+![flow](flow.png "Configuration processing")
+
+The user will configure pins and /or interconnections in the PtpConfig (hardware plugin section) as described above.
+If internal delay overrides are required, they can be configured together with the interconnections, as per this example:
+
+```yaml
+e810:
+	internalDelays:
```

Comment:
Prefer spaces over tabs for yaml indentation

Reply:
Fixed